src/main/java/sc/fiji/bdvpg/bdv/navigate/AdaptedViewCenterTransformCreator.java

```diff
@@ -0,0 +1,65 @@
+package sc.fiji.bdvpg.bdv.navigate;
+
+import bdv.util.BdvHandle;
+import net.imglib2.realtransform.AffineTransform3D;
+
+public class AdaptedViewCenterTransformCreator
+{
+	private final BdvHandle bdvHandle;
+	private final double[] xyz;
+	private final int t;
+
+	/**
+	 * Based on the current ViewerTransform, creates a new viewer transform with a new center position.
+	 *
+	 * @param bdvHandle
+	 * @param xyz
+	 * 			target coordinates in physical units
+	 * @param t
+	 * 			time point
+	 */
+	public AdaptedViewCenterTransformCreator( BdvHandle bdvHandle, double[] xyz, int t )
+	{
+		this.bdvHandle = bdvHandle;
+		this.xyz = xyz;
+		this.t = t;
+	}
+
+	private AffineTransform3D createAdaptedTransform()
+	{
+		bdvHandle.getViewerPanel().setTimepoint( t );
+
+		final AffineTransform3D currentViewerTransform = new AffineTransform3D();
+		bdvHandle.getViewerPanel().state().getViewerTransform( currentViewerTransform );
+
+		AffineTransform3D adaptedViewerTransform = currentViewerTransform.copy();
+
+		// ViewerTransform notes:
+		// - applyInverse: coordinates in viewer => coordinates in image
+		// - apply: coordinates in image => coordinates in viewer
+
+		final double[] targetPositionInViewerInPixels = new double[ 3 ];
+		currentViewerTransform.apply( xyz, targetPositionInViewerInPixels );
+
+		for ( int d = 0; d < 3; d++ )
+		{
+			targetPositionInViewerInPixels[ d ] *= -1;
+		}
+
+		adaptedViewerTransform.translate( targetPositionInViewerInPixels );
+
+		final double[] windowCentreInViewerInPixels = new double[ 3 ];
+		windowCentreInViewerInPixels[ 0 ] = bdvHandle.getViewerPanel().getDisplay().getWidth() / 2.0;
+		windowCentreInViewerInPixels[ 1 ] = bdvHandle.getViewerPanel().getDisplay().getHeight() / 2.0;
+
+		adaptedViewerTransform.translate( windowCentreInViewerInPixels );
+
+		return adaptedViewerTransform;
+	}
+
+	public AffineTransform3D getAdaptedCenterTransform()
```

Comment:
I would be in favor of adding a Run function like the other actions. The run function would center the BdvHandle to the given coordinates. Also it would be nice if you can give an Array of BdvHandle in order to recenter a set of BdvHandle.

The AffineTransform could be returned with a getter method

Reply:
OK, I see what you mean. 

But then this method also needs the `animationDurationTime` input parameter of the `ViewerTransformChanger`.
Feels a bit like not so good separation of concerns to me.
In this sense keeping methods that _create_ viewer transforms (there could be many) and that _apply_ the viewer transform (maybe we have only one, but there could be also more in the future) separate feels better to me. What do you think?

I think the only reason one would need just one class that does both would be for the UI, and there I think we anyway would need to create a `Command` that then would wrap two calls like this: `new TransformApplier( new TransformCreator.get() ).run()`, isn't it?

So, in other words, I would prefer to put directly executable actions into `Command`s and keep things more separate on the backend API side.

Reply:
Ok, then why not BdvHandleHelper (this class may already exists) which gives  a few useful static function (centerViewOn, rescale, etc.) and only one or few actions which apply the transform on one or several BdvHandles ?